.github/workflows/test.yml

```diff
@@ -86,6 +86,8 @@ jobs:
         if: github.event_name == 'push' && github.ref == 'refs/heads/main' && steps.check_conan_cache.outputs.all_in_cache != 'true'
         run: |
           rm -rf ./.conan
+          # Re-add conan deps after flushing the conan cache before building packages.
```

Comment:
We need to update the cache when the dependencies change. Otherwise it will grow unbounded and there are only 10GB available in Github Actions

Reply:
We need `./conan_deps` to verify that the the package is present (otherwise it won't be found). We then remove the cache if any are outdated because we don't want the new cache to contain the old and the new. This removes the deps we just added so we need to readd it before building with the new cache (this sequence will only be triggered on the main branch when pushing and even then only if the cache is not fully valid).